### PR TITLE
 support show create table syntax

### PR DESCRIPTION
**Pull request by siddontang**
support `show create table t `

### CODE REVIEW

File: plan/plans/show.go [hunk 1]
```diff
@@ -379,3 +394,68 @@ func (s *ShowPlan) fetchShowDatabases(ctx context.Context) error {
...
+		buf.WriteString(fmt.Sprintf("  `%s` %s", col.Name.O, col.FieldType.String()))
+		if col.DefaultValue == nil {
+			buf.WriteString(" DEFAULT NULL")
+		} else {
+			buf.WriteString(fmt.Sprintf(" DEFAULT %v", col.DefaultValue))
+		}
+		if i != len(tb.Cols())-1 {
+			buf.WriteString(",\n")
+		} else {
+			if len(tb.Indices()) == 0 {
```
Comment (qiuyesuifeng) [inline plan/plans/show.go]: How about trimming last line `\n` instead of add check here?

Comment (qiuyesuifeng) [inline plan/plans/show.go]: I mean replace last `,\n` with `\n`, then needless to check?

Comment (siddontang) [inline plan/plans/show.go]: no, must use `,\n`

File: plan/plans/show.go [hunk 2]
```diff
@@ -182,17 +188,26 @@ func (s *ShowPlan) evalCondition(ctx context.Context, m map[interface{}]interfac
 	return expression.EvalBoolExpr(ctx, cond, m)
 }
 
-func (s *ShowPlan) fetchShowColumns(ctx context.Context) error {
+func (s *ShowPlan) getTable(ctx context.Context) (table.Table, error) {
```
Comment (ngaut) [inline plan/plans/show.go]: @coocood

**Comment by siddontang** [comment]
PTAL @shenli

File: plan/plans/show_test.go [hunk 3]
```diff
@@ -63,6 +63,23 @@ func (p *testShowSuit) SetUpSuite(c *C) {
 	variable.BindSessionVars(p)
 }
 
+func (p *testShowSuit) TestCommon(c *C) {
```
Comment (coocood) [inline plan/plans/show_test.go]: This test tests several types of show plans, and they are not common for all show plans.
I think rename it to `TestSimple` would be better.

File: plan/plans/show.go [hunk 4]
```diff
@@ -379,3 +394,80 @@ func (s *ShowPlan) fetchShowDatabases(ctx context.Context) error {
...
+			buf.WriteString(fmt.Sprintf("  UNIQUE KEY `%s` ", idx.Name.O))
+		} else {
+			buf.WriteString(fmt.Sprintf("  KEY `%s` ", idx.Name.O))
+		}
+
+		cols := make([]string, 0, len(idx.Columns))
+		for _, c := range idx.Columns {
+			cols = append(cols, c.Name.O)
+		}
+		buf.WriteString(fmt.Sprintf("(`%s`)", strings.Join(cols, "`,`")))
```
Comment (shenli) [inline plan/plans/show.go]: if len(cols) == 1, parentheses are not necessary.

Comment (siddontang) [inline plan/plans/show.go]:
```
mysql> show create table t;
+-------+----------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                             |
+-------+----------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `c1` int(11) DEFAULT NULL,
  KEY `c1` (`c1`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1 |
```

should need.

**Comment by shenli** [comment]
LGTM

**Pull request merged**